plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/helpers.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs-extra';
+import { resolve as resolvePath, isAbsolute } from 'path';
+import { InputError, UrlReader } from '@backstage/backend-common';
+import { ScmIntegrations } from '@backstage/integration';
+import { JsonValue } from '@backstage/config';
+
+export async function fetchContents({
+  reader,
+  integrations,
+  baseUrl,
+  fetchUrl = '.',
+  outputPath,
+}: {
+  reader: UrlReader;
+  integrations: ScmIntegrations;
+  baseUrl?: string;
+  fetchUrl?: JsonValue;
+  outputPath: string;
+}) {
+  if (typeof fetchUrl !== 'string') {
+    throw new InputError(
+      `Invalid url parameter, expected string, got ${typeof fetchUrl}`,
+    );
+  }
+
+  let fetchUrlIsAbsolute = false;
+  try {
+    // eslint-disable-next-line no-new
+    new URL(fetchUrl);
+    fetchUrlIsAbsolute = true;
+  } catch {
+    /* ignored */
+  }
+
+  // We handle both file locations and url ones
+  if (!fetchUrlIsAbsolute && baseUrl?.startsWith('file://')) {
+    const basePath = baseUrl.slice('file://'.length);
```

Comment:
what the heck, string.slice instead of substring? :) didn't know this was a thing

Reply:
It is :p